presto-docs/src/main/sphinx/security/internal-communication.rst

```diff
@@ -0,0 +1,80 @@
+=============================
+Secure Internal Communication
+=============================
+
+The Presto cluster can be configured to use secured communication. Communication
+between Presto nodes can be secured with encryption, authentication and authorization.
+
+.. Note: At this step only encryption is supported
+
+Encrypted Internal Communication Configuration
+----------------------------------------------
+
+Encryption is configured in `config.properties` file.
+Encryption properties are identical for all of the presto node roles (coordinator, worker)
+and must be set on every node in the cluster.
+
+To enable encryption for the Presto internal communication follow the next steps:
+
+1. Disable HTTP endpoint.
+
+.. code-block:: none
+
+    http-server.http.enabled=false
+
+.. warning::
+
+    Leaving HTTP endpoint enabled leads to a security hole.
+
+2. Specify node's domains. Every node must have it's own domain configured.
+Domains must be in the same parent domain to be able to easy generate custom SSL
+certificates. e.g.: `coordinator.example.com`, `worker1.example.com`, `worker2.example.com`.
+
+.. code-block:: none
+
+    node.external-address=<node-domain>
+
+3. Generate Java Keystore File.
+
+.. code-block:: none
+
+    keytool -genkeypair -alias example.com -keyalg RSA -keystore keystore.jks
+    Enter keystore password:
+    Re-enter new password:
+    What is your first and last name?
+      [Unknown]:  *.example.com
```

Comment:
It is literally. Every presto node must be able to connect to any other node within the same cluster. We can create private certificates for every single node for the exact domain, and than specify it for server, and create a keystore that has all the public keys for all the domains, and specify it for client. But for the simplicity we suggest to just create one single keystore for the base domain with `*`, and distribute it across the cluster.